orca-bakery/src/main/groovy/com/netflix/spinnaker/orca/bakery/tasks/MonitorBakeTask.groovy

```diff
@@ -48,16 +48,16 @@ class MonitorBakeTask implements OverridableTimeoutRetryableTask {
     def previousStatus = stage.context.status as BakeStatus
 
     try {
-      def newStatus = bakery.lookupStatus(region, previousStatus.id).toBlocking().single()
+      def newStatus = bakery.lookupStatus(region, previousStatus.id)
       if (isCanceled(newStatus.state) && previousStatus.state == BakeStatus.State.PENDING) {
         log.info("Original bake was 'canceled', re-baking (executionId: ${stage.execution.id}, previousStatus: ${previousStatus.state})")
         def rebakeResult = createBakeTask.execute(stage)
         return new TaskResult(ExecutionStatus.RUNNING, rebakeResult.context, rebakeResult.outputs)
       }
 
       new TaskResult(mapStatus(newStatus), [status: newStatus])
-    } catch (RetrofitError e) {
-      if (e.response?.status == 404) {
+    } catch (HttpException e) {
```

Comment:
We'll something similar to the `RetrofitExceptionHandler` (https://github.com/spinnaker/orca/blob/master/orca-retrofit/src/main/groovy/com/netflix/spinnaker/orca/retrofit/exceptions/RetrofitExceptionHandler.groovy) that speaks `HttpException` and whatever we wrap network errors with.